frontend/src/ui-components/AcmTable/AcmTable.tsx

```diff
@@ -745,12 +745,15 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
         .map((col) => col.id as string),
     [columns]
   )
-  const localSavedCols = JSON.parse(localStorage.getItem(id + 'SavedCols')!)
-  const localSavedColOrder = JSON.parse(localStorage.getItem(id + 'SavedColOrder')!)
-  const [colOrderIds, setColOrderIds] = useState<string[]>(localSavedColOrder || defaultOrderIds)
+  const localSavedCols = JSON.parse(localStorage.getItem(id + 'SavedCols') || '[]')
+  const localSavedColOrder = JSON.parse(localStorage.getItem(id + 'SavedColOrder') || '[]')
+  const [colOrderIds, setColOrderIds] = useState<string[]>(
+    localSavedColOrder?.length > 0 ? localSavedColOrder : defaultOrderIds
+  )
```

Comment:
There are utility methods in this file that handle exceptions in JSON parsing (`setLocalStorage` and `getLocalStorage`)... I'm not sure why those were used initially, but could we make use of them here instead?

When we work with localStorage, we should always treat it as untrusted data and code defensively. I am able to crash the page by putting something that is not JSON into the local storage values.